### PR TITLE
wrapped pubsub message attributes method

### PR DESCRIPTION
**Pull request by LironKS**

### CODE REVIEW

File: instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java [hunk 1]
```diff
@@ -98,4 +104,25 @@ public static void buildAndFinishSpan(Context context, PubsubMessage pubsubMessa
     Context current = subscriberInstrumenter.start(newContext, pubsubMessage);
     subscriberInstrumenter.end(current, pubsubMessage, null, null);
   }
+
+  public static Optional<Object> extractPubsubMessageAttributes(PubsubMessage pubsubMessage) {
+    try {
+      Class cls = pubsubMessage.getClass();
```
Comment (nozik) [inline instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java]: Why not create a "getAttributeWithReflection" method? Receives an object, attribute name?

**Review by nozik — CHANGES_REQUESTED** [review]

File: instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java (hunk 1 above)
Comment (LironKS) [reply instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java]: it's not that generic.. it's very specific for pubsubmessage

Comment (nozik) [reply instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java]: What about something like this?

`public static extractAttributeFromObject(Object object, String attrName) {
	Class cls = object.getClass();
	Field attribute = cls.getDeclaredField(attrName);
	attribute.setAccessible(true);
	return attribute.get(object);
}`

**Review by nozik — CHANGES_REQUESTED** [review]

File: instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java [hunk 2]
```diff
@@ -107,22 +107,39 @@ public static void buildAndFinishSpan(Context context, PubsubMessage pubsubMessa
...
+      }
     } catch (Exception e) {
-      System.out.println("Got Exception while instrumenting pubsubMessage: " + e);
-
+      return Optional.empty();
     }
     return Optional.empty();
   }
+
+  private static Field extractAttributeFromObject(Object object, String fieldName) {
```
Comment (nozik) [inline instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java]: Why not return the property value itself?

Comment (LironKS) [reply instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java]: I thought its clearer to get the object from outside, no ?

Comment (nozik) [reply instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java]: IMO this method is an abstraction for extracting the attribute, so we shouldn't be interested in reflection objects like Field, etc.

Comment (LironKS) [reply instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java]: ok. done.

File: instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java [hunk 3]
```diff
@@ -98,4 +104,38 @@ public static void buildAndFinishSpan(Context context, PubsubMessage pubsubMessa
     Context current = subscriberInstrumenter.start(newContext, pubsubMessage);
     subscriberInstrumenter.end(current, pubsubMessage, null, null);
   }
+
+  public static Optional<Object> extractPubsubMessageAttributes(PubsubMessage pubsubMessage) {
+
```
Comment (nozik) [inline instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java]: remove newline

File: instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java [hunk 4]
```diff
@@ -98,4 +104,38 @@ public static void buildAndFinishSpan(Context context, PubsubMessage pubsubMessa
...
+  }
+
+  private static Object extractAttributeFromObject(Object object, String fieldName) {
+    try {
+      Class cls = object.getClass();
+      Field field = cls.getDeclaredField(fieldName);
+      field.setAccessible(true);
+      return field.get(object);
+    } catch (Exception e) {
+      System.out.println(
```
Comment (nozik) [inline instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java:130]: Isn't there a logger for OTEL that can be used?

**Review by nozik — APPROVED** [review]

File: instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java (hunk 4 above)
Comment (LironKS) [reply instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java:130]: I didn't see something similar in the other instrumentations

**Pull request merged**